src/util.c

```diff
@@ -1445,16 +1445,16 @@ ni_bool_t
 ni_file_exists_fmt(const char *fmt, ...)
 {
 	char *path = NULL;
-	ni_bool_t ret;
 	va_list ap;
+	int ret;
 
 	if (!fmt)
 		return FALSE;
 
 	va_start(ap, fmt);
 	ret = vasprintf(&path, fmt, ap) > 0;
 	va_end(ap);
-	if (!ret)
+	if (ret < 0 || !path)
```

Comment:
maybe it's time to switch to `ni_string_printf()` instead...

Reply:
This wouldn't work but require a `ni_string_vprintf` not `ni_string_printf`. The intention was to fix a warning/bug, not to refactor the functions.

Reply:
ack